src/TypeScript/Analyzer/EslintBridgeIssueConverter.cs

```diff
@@ -32,25 +32,20 @@ internal interface IEslintBridgeIssueConverter
         IAnalysisIssue Convert(string filePath, Issue issue);
     }
 
-    internal delegate string ConvertToSonarRuleKey(string eslineRuleKey);
-    internal delegate IEnumerable<RuleDefinition> GetRuleDefinitions();
-
     internal class EslintBridgeIssueConverter : IEslintBridgeIssueConverter
     {
-        private readonly ConvertToSonarRuleKey keyMapper;
-        private readonly GetRuleDefinitions ruleDefinitionProvider;
+        private readonly IRulesProvider rulesProvider;
 
-        public EslintBridgeIssueConverter(ConvertToSonarRuleKey keyMapper, GetRuleDefinitions ruleDefinitionProvider)
+        public EslintBridgeIssueConverter(IRulesProvider rulesProvider)
```

Comment:
The converter now consumes the common interface rather than separate delegates.

I was thinking about the `GetSonarRuleKey` method: perhaps it should just be a method on this class, rather than being part of the interface. Still, it would be easy to move later if necessary.